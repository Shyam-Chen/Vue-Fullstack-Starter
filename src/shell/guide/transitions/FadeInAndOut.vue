<template>
  <v-layout column class="my-1">
    <div class="title mb-2">Fade In and Out</div>

    <v-layout align-center>
      <v-btn @click="fade.transition = !fade.transition">Fade Transition</v-btn>

      <transition name="fade-transition">
        <div v-if="fade.transition">Block</div>
      </transition>
    </v-layout>

    <v-layout align-center class="my-1">
      <v-btn @click="fade.animation = !fade.animation">Fade Animation</v-btn>

      <transition name="fade-animation">
        <div v-if="fade.animation">Block</div>
      </transition>
    </v-layout>
  </v-layout>
</template>

<script>
// @flow

type Data = {
  fade: { transition: boolean, animation: boolean },
};

export default {
  data(): Data {
    return {
      fade: {
        transition: true,
        animation: true,
      },
    };
  },
};
</script>

<style scoped>
/* fade-transition */
.fade-transition-enter-active,
.fade-transition-leave-active {
  transition: opacity 0.5s;
}

.fade-transition-enter,
.fade-transition-leave-to {
  opacity: 0;
}

/* fade-animation */
.fade-animation-enter-active {
  animation: fade 0.5s;
}

.fade-animation-leave-active {
  animation: fade 0.5s reverse;
}

@keyframes fade {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}
</style>
