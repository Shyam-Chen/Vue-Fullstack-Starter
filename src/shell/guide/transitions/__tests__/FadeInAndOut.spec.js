import { shallowMount } from '@vue/test-utils';

import FadeInAndOut from '../FadeInAndOut';

describe('FadeInAndOut', () => {
  let [wrapper] = [];

  beforeEach(() => {
    wrapper = shallowMount(FadeInAndOut);
  });

  it('should render an initial component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
