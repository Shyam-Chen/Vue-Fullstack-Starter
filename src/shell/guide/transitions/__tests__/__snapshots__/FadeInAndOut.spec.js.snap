// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeInAndOut should render an initial component 1`] = `
<v-layout-stub tag="div" column="" class="my-1">
  <div class="title mb-2">Fade In and Out</div>
  <v-layout-stub tag="div" align-center="">
    <v-btn-stub tag="button" activeclass="" type="button">Fade Transition</v-btn-stub>
    <transition-stub name="fade-transition">
      <div>Block</div>
    </transition-stub>
  </v-layout-stub>
  <v-layout-stub tag="div" align-center="" class="my-1">
    <v-btn-stub tag="button" activeclass="" type="button">Fade Animation</v-btn-stub>
    <transition-stub name="fade-animation">
      <div>Block</div>
    </transition-stub>
  </v-layout-stub>
</v-layout-stub>
`;
